compare_hashes/score.h

```diff
@@ -0,0 +1,60 @@
+#ifndef SCORE_H
+#define SCORE_H
+
+#include <cassert>
+
+typedef int location_in_submission;
+typedef unsigned int hash;
+typedef std::string user_id;
+typedef unsigned int version_number;
+
+// represents the plagiarism score for a given submissions, used for the overall rankings file
+class Score {
+public:
+  // CONSTRUCTOR
+  Score(unsigned int hashes_matched, float percent): hashes_matched(hashes_matched), percent(percent), score(-1) {}
+  Score(const Score &other) { copy(other); }
+
+  // GETTERS
+  float getPercent() const { return percent; }
+  unsigned int getHashesMatched() const { return hashes_matched; }
+
+  // MODIFIER
+  // Each submission in the ranking file gets a composite score that weighs both its percentage
+  // of suspicious matches, and its percentile of total number of hashes matched
+  void calculateScore(unsigned int max_hashes_matched) {
+    assert(PERCENT_WEIGHT + MATCH_WEIGHT == 1);
+    score = PERCENT_WEIGHT*(percent/100.0) + MATCH_WEIGHT*((1.0*hashes_matched)/max_hashes_matched);

```

Comment:
No need for the `1.0*hashes_matched`, upcasts are done implicitly in c++ and multiplying by 1 does nothing in this case

Reply:
I am not really sure what you mean here.  I created a simple C++ program and this does not seem to be correct unless I'm misunderstanding what you mean.

```
    int x = 10;
    int y = 3;
    std::cout << "x/y: " << x/y << std::endl;
    std::cout << "(1.0*x)/y: " << (1.0*x)/y << std::endl;
```
Output:
```
x/y: 3
(1.0*x)/y: 3.33333
```

Reply:
Oh my bad I thought one of those was a float already which would have caused an upcast on the other,

Use `static_cast<float>(foo)` to convert an u_int to a float